### DISCUSSION TITLE
How to get bonds type with mol generated by Chem.MolFromXYZBlock(*) ?

### DISCUSSION THREAD

**Discussion by ShangChien** [rdkit/rdkit#5759] [Q&A] [answered]
It is amazing for the feature of getting mol-object by function of `Chem.MolFromXYZBlock()`. 
But, it does not contain a bond relationship. My project need to inspect the bond type between atoms.
So ,how to calculate (add or map) bond infomation with a mol-object generated by Chem.MolFromXYZBlock(*) ?
### **current:**
_conformer structure optimized by xtb, output is XYZfile:_

![image](https://user-images.githubusercontent.com/48350636/201817061-a258cb7a-a403-49c6-8376-631304f17ca8.png)

### **expect:**
_with kekule bond-system: single-bond, double-bond and triple-bond_

![image](https://user-images.githubusercontent.com/48350636/201817353-a2fafaae-5cee-4ab4-b623-0cfc1a70aa33.png)

**Comment by greglandrum** [comment, marked as answer]
Hi @ShangChien. In the most recent version of the rdkit (v2022.09.1) @gosreya  added an implementation of @janholstjensen's xyz2mol to the RDKit core to go along with `MolFromXYZBlock()`

I haven't done a blog post on this yet, but there's a bit of a demo in my notebook from this year's RDKit UGM: https://github.com/rdkit/UGM_2022/blob/main/Notebooks/Landrum_WhatsNew.ipynb

The basic idea is that there are two functions you can use: 
1. `rdDetermineBonds.DetermineConnectivity()`, which determines which atoms should be connected via bonds 
2. `rdDetermineBonds.DetermineBonds()`, which determines both which atoms should be connected and what the bond orders are.

Note that both functions need the overall charge on the molecule.

**Reply by ShangChien** [reply]
So perfect !
Thank you !

**Comment by ShangChien** [comment]
![image](https://user-images.githubusercontent.com/48350636/201836473-1c8dd79d-5a07-4452-a8b6-8e05eccd8101.png)
 @greglandrum Maybe it's not ready for public. I'm happy to wait it.

**Reply by greglandrum** [reply]
How did you install the RDKit?

**Reply by ShangChien** [reply]
I directly install by `pip install rdkit` on my Ubuntu, and I inspected the file of ../rdkit/Chem, didn't find file named rdDetermineBonds
> How did you install the RDKit?

**Comment by greglandrum** [comment]
@kuelumbus : have you enabled `RDK_BUILD_XYZ2MOL_SUPPORT` in the RDKit pypi builds?

**Reply by kuelumbus** [reply]
Yes, it's enabled but the builds are not yet available at PyPi. I was thinking of waiting for you to release the next RDKit tag but I can also upload a version `rdkit-pypi 2022.9.1.1` that has XYZ2MOL_SUPPORT. When is the next RDKit release due?

**Reply by greglandrum** [reply]
The next patch release should be next week

**Reply by ShangChien** [reply]
Thank all of your contributions for providing a very simple installation of rdkit  via channel of pypi.
